### DISCUSSION TITLE
Customized model for Thai language

### DISCUSSION THREAD

**Discussion by GoldenBigAnt** [PaddlePaddle/PaddleOCR#14471] [General]
I made Thai lang model. but paddleocr don't support the Thai lang yet. how can I use directly the trained thai model? please help me

**Comment by GreatV** [comment]
To use your trained Thai language model in PaddleOCR, even though Thai is not officially supported yet, you can follow these steps:

### 1. **Update the YAML Configuration**
   - PaddleOCR uses YAML files to configure the training and inference process. Since you’ve already trained a Thai model, ensure that the YAML file you used during training is properly configured to include:
     - The Thai dataset path.
     - The Thai character set (`character_dict_path`), which should include all Thai characters you want to recognize.
     - Model architecture and hyperparameters suitable for Thai scripts.
   - Example YAML changes:
     ```yaml
     Global:
       character_dict_path: "path_to_your_thai_character_list.txt"
     ```

   - If you’ve used a YAML configuration for English or another language, modify the `character_dict_path` to point to your Thai character list.

---

### 2. **Ensure Thai Dataset Preprocessing**
   - If your Thai model was trained on a dataset, make sure the dataset was preprocessed correctly. PaddleOCR uses the `tools/create_lmdb_dataset.py` script to preprocess text datasets into LMDB format.
   - For Thai, ensure the text annotations are encoded in Unicode to handle the complex script.

---

### 3. **Inference with a Custom Model**
   - During inference, you need to specify the trained Thai model and its corresponding configuration YAML file. You can run the following command to perform inference:
     ```bash
     python tools/infer_rec.py --config_path="path_to_your_yaml_config.yaml" --image_dir="path_to_image" --rec_model_dir="path_to_trained_thai_model/"
     ```
   - Replace `path_to_your_yaml_config.yaml` with your Thai YAML configuration file and `path_to_trained_thai_model/` with the directory containing your trained Thai model.

---

### 4. **Update the Dictionary for Thai**
   - PaddleOCR uses a character dictionary for recognition. Make sure your Thai character dictionary (`character_dict_path`) is correctly specified and matches the characters your model was trained on.
   - Example Thai character list (save as `thai_dict.txt`):
     ```
     ก
     ข
     ฃ
     ค
     ...
     (all Thai characters)
     ```

---

### 5. **Debugging Issues**
   If your model isn't working as expected or PaddleOCR defaults to a different language (like English), check the following:
   - Ensure you are pointing to the correct `.yaml` file during inference.
   - Validate that the trained weights (`.pth` or `.pdparams` file) are correctly linked in the YAML file.
   - Ensure the character dictionary aligns with the model's training.

---

### 6. **Multilingual Support**
   If you want PaddleOCR to support both Thai and other languages, you can create a combined multilingual character list that includes characters from all desired languages. Adjust the `character_dict_path` and retrain or fine-tune the model with this multilingual dataset.

   **Reference**: There’s a related discussion [here](https://github.com/PaddlePaddle/PaddleOCR/discussions/14114) about fine-tuning a model for Indonesian. While it focuses on Indonesian, the process for Thai is similar:
   - Fine-tune with a Thai-specific dataset.
   - Adjust the YAML configuration for Thai.
   - Use the proper dictionary for Thai characters.

---

By following these steps, you should be able to use your trained Thai language model in PaddleOCR. If you encounter specific errors, sharing the error messages can help diagnose the problem further.

---
*Response generated by [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*

**Reply by GoldenBigAnt** [reply]
thanks for your help. I followed above steps and made just best_accuracy model already. 
how can I use the model in python?

**Comment by GoldenBigAnt** [comment]
![Screenshot_1](https://github.com/user-attachments/assets/55b7363d-63b1-482c-aeb7-668e7e54793f)

![Uploading Screenshot_1.png…]()

**Reply by BlackHand133** [reply]
How can you fix missing characters or punctuation marks in the OCR output?

**Comment by GoldenBigAnt** [comment]
please see, I can get this result from own trained model
but I can't know how can I use the model for other images. please help me

**Comment by jingsongliujing** [comment]
You can see this document：https://paddlepaddle.github.io/PaddleOCR/latest/en/ppocr/infer_deploy/python_infer.html#text-detection-model-inference

**Comment by GoldenBigAnt** [comment]
Happy new year.
I checked above link, but I can get good result only one line jpg file. but I have to ocr for mutiline jpg files.
![ค่าเช่าอาคารสำนักงาน ลิซ่า แอนด์ ลิน่า](https://github.com/user-attachments/assets/e9b5d328-c7df-458f-89f1-355b721e59db)
like this file.
I used "python3 tools/infer/predict_rec.py --image_dir="./doc/imgs_words/korean/1.jpg" --rec_model_dir="./your inference model" --rec_char_dict_path="ppocr/utils/dict/korean_dict.txt" --vis_font_path="doc/fonts/korean.ttf"" this command.
but it can get only one line.
please help me asap.

**Comment by cuicheng01** [comment]
Feel free to submit your request directly in this issue:
👉 [PaddleOCR Multilingual Support Request #15617](https://github.com/PaddlePaddle/PaddleOCR/issues/15617)

**Comment by gheorghebg11** [comment]
@GoldenBigAnt Would you share your trained model ?
